grpc/src/main/java/com/linecorp/armeria/server/grpc/TextUnframedGrpcErrorHandler.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2024 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.grpc;
+
+import static com.linecorp.armeria.server.grpc.UnframedGrpcErrorHandlerUtil.responseCause;
+import static com.linecorp.armeria.server.grpc.UnframedGrpcErrorHandlerUtil.withDefault;
+
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpData;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.grpc.protocol.GrpcHeaderNames;
+import com.linecorp.armeria.common.util.Exceptions;
+import com.linecorp.armeria.internal.common.util.TemporaryThreadLocals;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+import io.grpc.Status;
+import io.grpc.Status.Code;
+
+/**
+ * Error handler which maps a gRPC response to an {@link HttpResponse} in plaintext format.
+ */
+final class TextUnframedGrpcErrorHandler implements UnframedGrpcErrorHandler {
+    private final UnframedGrpcStatusMappingFunction statusMappingFunction;
+    private static final UnframedGrpcStatusMappingFunction DEFAULT_STATUS_MAPPING_FUNCTION =
+            UnframedGrpcStatusMappingFunction.of();
+    private static final TextUnframedGrpcErrorHandler DEFAULT =
+            new TextUnframedGrpcErrorHandler(DEFAULT_STATUS_MAPPING_FUNCTION);
+
+    private TextUnframedGrpcErrorHandler(UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        this.statusMappingFunction = withDefault(statusMappingFunction);
+    }
+
+    static TextUnframedGrpcErrorHandler of() {
+        return DEFAULT;
+    }
+
+    static TextUnframedGrpcErrorHandler of(UnframedGrpcStatusMappingFunction statusMappingFunction) {
+        if (DEFAULT_STATUS_MAPPING_FUNCTION.equals(statusMappingFunction)) {
```

Comment:
Reference checking would be sufficient. 
```suggestion
        if (DEFAULT_STATUS_MAPPING_FUNCTION == statusMappingFunction) {
```